reposerver/repository/repository.go

```diff
@@ -300,6 +300,7 @@ func (s *Service) runRepoOperation(
 	var gitClient git.Client
 	var helmClient helm.Client
 	var err error
+	gitClientOpts := git.WithCache(s.cache, !settings.noRevisionCache && !settings.noCache)
```

Comment:
It seems that the `repository.Service.cache` is just Redis based and doesn't leverage the `twolevel` cache client. Can you please confirm my understanding of this part of the code? If so, it means that we are moving the chatty communication from Git to Redis which can bring new problems.

I don't want to push for a premature optimization as long as we are aware of this possible behaviour and have a plan B if we want to go with this implementation which I am not opposed to.

Reply:
That's right, overall redis requests are down after the change due to not having to try and get and set the values for each application each time. 

I do however have the optimization for moving this to a two level cache as part of the original PR and in the final part of the PR split https://github.com/nromriell/argo-cd/pull/3